addons/website/models/ir_http.py

```diff
@@ -245,6 +245,15 @@ def _handle_exception(cls, exception):
             if not request.uid:
                 cls._auth_method_public()
 
+            # we rollback the current transaction to avoid deadlocks
+            # to fully explain the issue, if the current transaction is holding a lock
+            # creating a new cursor that might wait for the lock to be freed.
+            # This will happen only after the request is done.
+            # Which means, we have a thread/worker frozen until its timeout is attained.
+            # So rollbacking the transaction will free the locks and since we are in a case
+            # where an exception is raised, it means that the transaction shouldn't be committed
```

Comment:
If I may, here's a proposed rewording. Message unaltered.
```py
            # We rollback the current transaction before initializing a new
            # cursor to avoid potential deadlocks.

            # If the current (failed) transaction was holding a lock, the new
            # cursor might have to wait for this lock to be released further
            # down the line. However, this will only happen only after the
            # request is done. As a result, the current thread/worker is frozen
            # until its timeout is reached.

            # So rolling back the transaction will release any potential lock
            # and, since we are in a case where an exception was raised, the
            # transaction shouldn't be committed in the first place.
```